### PR TITLE
Enable js unit test files to live alongside implementation files

### PR DESCRIPTION
**Pull request by gusaiani**
By allowing unit test files to live in the same directory as the implementation files, we want to encourage engineers to write and maintain unit test files.

This should:

* reduce regressions
* encourage modularization
* balance reliance on fast-running unit tests versus slow-running integration tests

Please see a [couple of early positive reactions to this idea](https://www.notion.so/56f6f59da3b647de89acb28ea9ebc424?v=cca33933b6244a7a94f256d1c110ce09&p=5e3272c9ec0e472493c7d0e1b9bd5cb3).

### After 

In this branch, run

```
yarn test-unit frontend/src/metabase/dashboard/components/DashCard.unit.spec.js
```

You should see the following output:

```
Test Suites: 1 skipped, 0 of 1 total
Tests:       1 skipped, 1 total
Snapshots:   0 total
Time:        4.348s
Ran all test suites matching "frontend/src/metabase/dashboard/components/DashCard.unit.spec.js".
```

### Before

Create a throwaway branch off of `master`, as the situation below needs to be created.

```
git checkout -b throwaway-branch
mv frontend/test/metabase/dashboard/DashCard.unit.spec.js frontend/src/metabase/dashboard/components
yarn test-unit frontend/src/metabase/dashboard/components/DashCard.unit.spec.js
```

You will get a **No tests found** — jest is not configured to look inside `frontend/src` dir.

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
This sounds like a very good idea to me.
It's much easier to see what's the coverage even without running a single test.

Also, people will be more incentivized to write unit tests with this model.
I don't see any downsides, so it has 👍  from me.

File: jest.unit.conf.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
   },
   "testPathIgnorePatterns": ["<rootDir>/frontend/test/.*/.*.tz.unit.spec.js"],
   "testMatch": [
-    "<rootDir>/frontend/test/**/*.unit.spec.js?(x)",
-    "<rootDir>/enterprise/frontend/test/**/*.unit.spec.js?(x)"
+    "<rootDir>/frontend/**/*.unit.spec.js?(x)",
+    "<rootDir>/enterprise/frontend/**/*.unit.spec.js?(x)"
```
Comment (nemanjaglumac) [inline jest.unit.conf.json:10]:
```suggestion
    "<rootDir>/**/*.unit.spec.js?(x)",
```

If we're going to apply this approach, then it shouldn't even matter does the test live in oss or ee folder.

Comment (gusaiani) [reply jest.unit.conf.json]: @nemanjaglumac Totally. Thank you.

**Comment by gusaiani** [comment]
Closed as this change was embedded in https://github.com/metabase/metabase/pull/16966

**Pull request closed without merging**